src/main/java/com/example/java/maven/ticketVendingMachine/TicketVendingMachine.java

```diff
@@ -0,0 +1,115 @@
+package com.example.java.maven.ticketVendingMachine;
+
+
+public class TicketVendingMachine {
+    private final static String ASK_FOR_ANOTHER_TICKET = "Do you need another ticket? Insert Y for YES or N for NO.";
+    private final static String WRONG_ANSWER = "Wrong answer! Insert Y for YES or N for NO.";
+    private final static String ASK_FOR_TICKET = "Which ticket would you like to buy? Available tickets are: " + Ticket.getAllTicketSymbols();
+    private final static String INSERT_COIN = "Insert coin. Accepted coins are: " + Coin.getAllCoinSymbols();
+    private final static String GIVE_ODD_MONEY = "Here is your change";
+    private final static String VALUE_OF_TICKETS = "Value of your tickets is: ";
+
+
+
+    private final TicketPrinter ticketPrinter;
+    private final TicketStorage ticketStorage;
+    private final CoinStorage mainCoinStorage;
+    private final CoinStorage tempCoinStorage;
+    private final CoinStorage oddMoneyStorage;
+    private final MessagePrinter messagePrinter;
+    private final UserInputProvider userInputProvider;
+
+    public TicketVendingMachine(TicketPrinter ticketPrinter,
+                                TicketStorage ticketStorage,
+                                CoinStorage mainCoinStorage,
+                                CoinStorage tempCoinStorage,
+                                CoinStorage oddMoneyStorage,
+                                MessagePrinter messagePrinter,
+                                UserInputProvider userInputProvider) {
+
+        this.ticketPrinter = ticketPrinter;
+        this.ticketStorage = ticketStorage;
+        this.mainCoinStorage = mainCoinStorage;
+        this.tempCoinStorage = tempCoinStorage;
+        this.oddMoneyStorage = oddMoneyStorage;
+        this.messagePrinter = messagePrinter;
+        this.userInputProvider = userInputProvider;
+    }
+
+    public void fillMainCoinStorage() {
+        for (Coin coin : Coin.values()) {
+            mainCoinStorage.getCoins().put(coin, 10);
+        }
+    }
+
+    public Coin getCoin() {
```

Comment:
IMHO - this method should be in `UserInputProvider` class

Reply:
Moved to `UserInputProvider`